db/migrate/20221015114100_create_annotations.rb

```diff
@@ -0,0 +1,15 @@
+class CreateAnnotations < ActiveRecord::Migration[7.0]
+  def change
+    create_table :annotations do |t|
+      t.references :medium, null: false, foreign_key: true
+      t.references :user, null: false, foreign_key: true
+      t.text :timestamp, null: false
+      t.text :comment
+      t.integer :category
+      t.boolean :visible_for_teacher
+      t.string :color
+
```

Comment:
You did not specify `null: false` for some of these vales, so is the rest of the code working fine if we have a `null` value somewhere in here?

Reply:
I added a constraint for `category`. Also `color` shouldn't be nil - but here I added a more restrictive regex constraint that also filters strings which don't represent a hexadecimal color. Therefore a non-null constraint might be overkill (still, I can add one if you like).
All the other field actually have to be `nil` sometimes, so I'll won't change anything here.

Reply:
- It'd be great, if you could add these kinds of migrations directly do the original migration as described in [this comment](https://github.com/MaMpf-HD/mampf/pull/533#pullrequestreview-1691457370). Note that I've changed this now with the new commit: 63c15eaf25629
- What about the `subcategory` integer. Is it valid for an annotation to not have a subcategory?
- Can you point me to the exact code where you filter the strings to check if it is a hexadecimal color? Is this done in the backend? I'd still advocate to have `null: false` here as well. Otherwise, what does it mean if an annotation in the database has `nil` for its color?
- Why doesn't `visible_for_teacher` have to be non-null?

Reply:
- `subcategory` must only be selected if you select `content` as category (actually I should include that condition into the backend). So for each other category, it's ok for that field to be nil.
- The String check is done in the `is_valid_color`-method in the `AnnotationsController` (used by create and update), l. 148. Actually, in this method I already filter out the nil-case, so we'll never be in the situation that we have a nil value in the db. But I'll add the additional restriction - doesn't do any harm and maybe, we'll change something later and then it's better to have it ;).
- I decided to allow `visible_for_teacher` to be `nil` (and use this in basically the same way as `false`) because this part of the feature can be deactivated (originally, I wanted to differ between "user is not allowed to choose an option" and "user is actively saying no"). One could add a non-nil restriction but then has to pass `false` as default value (either in the form or in the controller).